### PR TITLE
Added some functions for supporting the functioning of Archival Nodes.

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

File: src/libLookup/Lookup.cpp [hunk 1]
```diff
@@ -990,6 +994,10 @@ bool Lookup::ProcessGetTxBlockFromSeed(const vector<unsigned char>& message,
                           .GetHeader()
                           .GetBlockNum();
     }
+    else if (lowBlockNum == 0)
+    {
+        lowBlockNum = 1;
```
Comment (Gnnng) [inline src/libLookup/Lookup.cpp]: I was trying to understand the reason why `lowBlockNum` is overwritten to 1. Can you clarify a bit?

Comment (KaustubhShamshery) [reply src/libLookup/Lookup.cpp]: so, as the current convention goes (1,0) ---> gives you the latest blocknum, and giving a request (0,0) --> gives you all the blocks in the blockchain. so keeping that in mind, when lookup recv 0 as low blocknum, it starts to give you the blocknums starting from 0.

Comment (Gnnng) [reply src/libLookup/Lookup.cpp]: Okay, got it. Perhaps we can add some comments there together with the code.

File: src/libLookup/Lookup.cpp [hunk 2]
```diff
@@ -1133,6 +1143,76 @@ bool Lookup::ProcessGetTxBodyFromSeed(const vector<unsigned char>& message,
...
+
+    uint32_t i = 0;
+    for (const auto& shard : shards)
+    {
+        LOG_GENERAL(INFO, "Size of shard " << i << " " << shard.size());
+        i++;
+    }
+    lock_guard<mutex> g(m_mutexShards);
+
+    m_mediator.m_ds->m_shards = shards;
```
Comment (ansnunez) [inline src/libLookup/Lookup.cpp]: I think you can do a std::move here to be more efficient

**Pull request merged**